EKF/sideslip_fusion.cpp

```diff
@@ -0,0 +1,215 @@
+            Kfusion[16] = SK_BETA[0]*(P[16][0]*SK_BETA[5] + P[16][1]*SK_BETA[4] - P[16][4]*SK_BETA[1] + P[16][5]*SK_BETA[2] + P[16][2]*SK_BETA[6] + P[16][6]*SK_BETA[3] - P[16][3]*SK_BETA[7] + P[16][22]*SK_BETA[1] - P[16][23]*SK_BETA[2]);
+            Kfusion[17] = SK_BETA[0]*(P[17][0]*SK_BETA[5] + P[17][1]*SK_BETA[4] - P[17][4]*SK_BETA[1] + P[17][5]*SK_BETA[2] + P[17][2]*SK_BETA[6] + P[17][6]*SK_BETA[3] - P[17][3]*SK_BETA[7] + P[17][22]*SK_BETA[1] - P[17][23]*SK_BETA[2]);
+            Kfusion[18] = SK_BETA[0]*(P[18][0]*SK_BETA[5] + P[18][1]*SK_BETA[4] - P[18][4]*SK_BETA[1] + P[18][5]*SK_BETA[2] + P[18][2]*SK_BETA[6] + P[18][6]*SK_BETA[3] - P[18][3]*SK_BETA[7] + P[18][22]*SK_BETA[1] - P[18][23]*SK_BETA[2]);
+            Kfusion[19] = SK_BETA[0]*(P[19][0]*SK_BETA[5] + P[19][1]*SK_BETA[4] - P[19][4]*SK_BETA[1] + P[19][5]*SK_BETA[2] + P[19][2]*SK_BETA[6] + P[19][6]*SK_BETA[3] - P[19][3]*SK_BETA[7] + P[19][22]*SK_BETA[1] - P[19][23]*SK_BETA[2]);
+            Kfusion[20] = SK_BETA[0]*(P[20][0]*SK_BETA[5] + P[20][1]*SK_BETA[4] - P[20][4]*SK_BETA[1] + P[20][5]*SK_BETA[2] + P[20][2]*SK_BETA[6] + P[20][6]*SK_BETA[3] - P[20][3]*SK_BETA[7] + P[20][22]*SK_BETA[1] - P[20][23]*SK_BETA[2]);
+            Kfusion[21] = SK_BETA[0]*(P[21][0]*SK_BETA[5] + P[21][1]*SK_BETA[4] - P[21][4]*SK_BETA[1] + P[21][5]*SK_BETA[2] + P[21][2]*SK_BETA[6] + P[21][6]*SK_BETA[3] - P[21][3]*SK_BETA[7] + P[21][22]*SK_BETA[1] - P[21][23]*SK_BETA[2]);
+
+        } else {
+            for (int i = 16; i <= 21; i++) {
+                Kfusion[i] = 0.0f;
+            }
+        }
+
+        // Calculate predicted sideslip angle and innovation using small angle approximation
+        _beta_innov = rel_wind(1) / rel_wind(0);
+
+		// Calculate the innovation variance
+		_beta_innov_var = 1.0f / SK_BETA[0];
+
+        // Compute the ratio of innovation to gate size
+		_beta_test_ratio = sq(_beta_innov) / (sq(fmaxf(_params.beta_innov_gate, 1.0f)) * _beta_innov_var);
+
+		// if the innovation consistency check fails then don't fuse the sample and indicate bad beta health
+		if (_beta_test_ratio > 1.0f) {
+			_beta_healthy = false;
+			return;
+		}
+
+        // Syntetic sideslip measurement sample has passed check so record it
+        _time_last_beta_fuse = _time_last_imu;
+
+		// Fuse syntetic sideslip measurement
+		fuse(Kfusion, _beta_innov); //Why calculate angle error when it is always zero?
+
+		// update covariance matrix via Pnew = (I - KH)P = P - KHP
+		for (unsigned row = 0; row < _k_num_states; row++) {
+			for (unsigned column = 0; column < _k_num_states; column++) { // Here it will be a lot of zeros, should optimize that...
+				KH[row][column] = Kfusion[row] * H_BETA[column];
+			}
+		}
+
+		for (unsigned row = 0; row < _k_num_states; row++) {
+			for (unsigned column = 0; column < _k_num_states; column++) {
+				for (unsigned i = 0; i < _k_num_states; i++) { // Check if this is correct matrix multiplication!
+					KHP[row][column] += KH[row][i] * P[i][column];
+				}
+			}
+		}
+
```

Comment:
Add the check used by other fusion operations where the state update and covariance update is not performed and a fault is recorded if any diagonals of KHP are greater than the corresponding diagonals on P.